### DISCUSSION TITLE
How to improve PageSpeed Insights score?

### DISCUSSION THREAD

**Discussion by mrtkp9993** [quarto-dev/quarto-cli#2631] [Q&A]
Hello,

I have a Quarto generated website on [https://muratkoptur.com/MyDsProjects](https://muratkoptur.com/MyDsProjects).

It takes very low performance score on [PageSpeed Insights](https://pagespeed.web.dev/):

![screencapture-pagespeed-web-dev-report-2022-09-27-18_47_54](https://user-images.githubusercontent.com/17238910/192574170-fe96ef1b-10ed-4561-9319-55bcf7838d5d.png)

How can we improve performance?

**Comment by dragonstyle** [comment]
I haven't looked super closely at this, but I notice that the quarto.org site itself has a much higher score- I'm wondering whether a specific feature that you've enabled that we haven't might be hurting performance (for example, the cookie prompt). It might also be as simple as the size or number of images on pages, but without being able to drill into the PageSpeed insights I can't really know. If you can open some of the items just under the screenshots of various pages, we could get some additional insights into where that time is going.

**Comment by mrtkp9993** [comment]
Thanks for the answer, I attached the detailed report.

I removed cookie consent for testing, but there is no significant change.

For "serve images in next-gen formats", I'll optimize my images.

[PageSpeed Insights.pdf](https://github.com/quarto-dev/quarto-cli/files/9661916/PageSpeed.Insights.pdf)

**Comment by mrtkp9993** [comment]
I optimized images with webp, but score didn't increase much.

[PageSpeed Insights - images optimized.pdf](https://github.com/quarto-dev/quarto-cli/files/9662276/PageSpeed.Insights.-.images.optimized.pdf)

**Reply by dragonstyle** [reply]
It looks to me like there are few big drivers (you could confirm by temporarily removing these)

- hypothesis
- Google Doubleclick Ads
- JSDelivr CDN / text-html-full.js

I didn't see a ton that Quarto could do internally to improve things, unfortunately. If you see some thing, we'd be very interested in making those improvements...

**Reply by mrtkp9993** [reply]
Thanks for the answer. I'll try to optimize, if I'll find something useful, I'll share it.

**Reply by mrtkp9993** [reply]
1. Can we remove unused css (PurgeCSS)? 
2. Can we defer js/css?
